src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java

```diff
@@ -107,20 +108,25 @@ public SecurityContextRepository contextRepository(ReactiveAuthenticationManager
                     .map(caller ->
                     {
                         var grantedAuthority = new ArrayList<SimpleGrantedAuthority>();
-                        caller.getRole().map(role ->
-                                grantedAuthority.add(new SimpleGrantedAuthority("ROLE_".concat(role))));
+                        if (caller.isVerified()) {
+                            grantedAuthority.add(new SimpleGrantedAuthority("ROLE_VERIFIED"));
+                        }
+                        caller.getRole().map(role -> grantedAuthority.add(new SimpleGrantedAuthority("ROLE_".concat(role))));
                         return new UsernamePasswordAuthenticationToken(caller, token, grantedAuthority);
                     })
                     .map(SecurityContextImpl::new);
         }
 
         private Mono<SecurityContext> checkCentralRegistry(String token) {
             return centralRegistryTokenVerifier.verify(token)
-                    .map(caller ->
-                            new UsernamePasswordAuthenticationToken(
-                                    caller,
-                                    token,
-                                    new ArrayList<SimpleGrantedAuthority>()))
+                    .map(caller -> {
+                        var grantedAuthority = new ArrayList<SimpleGrantedAuthority>();
+                        grantedAuthority.add(new SimpleGrantedAuthority("ROLE_VERIFIED"));
```

Comment:
does not make sense to add role verified here.

Reply:
Yes sure.